mythril/laser/ethereum/instructions.py

```diff
@@ -775,11 +775,9 @@ def mload_(self, global_state):
             data = global_state.new_bitvec("mem[" + str(simplify(op0)) + "]", 256)
             state.stack.append(data)
             return [global_state]
-
         try:
+            state.mem_extend(offset, 32)
             data = util.concrete_int_from_bytes(state.memory, offset)
-        except IndexError:  # Memory slot not allocated
-            data = global_state.new_bitvec("mem[" + str(offset) + "]", 256)
         except TypeError:  # Symbolic memory
             data = state.memory[offset]
```

Comment:
Should this use a slice?

Reply:
```suggestion
            data = state.memory[offset:offset+32]
```

Reply:
unfortunately everything should be concatenated there right?Straight concatenation lead to some failing tests which raised some Z3 exception, it might be good to raise a separate issue here.What do you think

Reply:
Sure, go ahead and create that